src/js/buffer.js

```diff
@@ -122,24 +122,24 @@ Buffer.isBuffer = function(object) {
   return util.isBuffer(object);
 };
 
-
-// buffer.equals(otherBuffer)
-Buffer.prototype.equals = function(otherBuffer) {
+//Equals and Compare methods checker
+function bufPrototypeCompare (otherBuffer, equalsOrCompare) {
   if (!util.isBuffer(otherBuffer)) {
-    throw new TypeError('Bad arguments: buffer.equals(Buffer)');
+    throw new TypeError('Bad arguments: buffer.' + equalsOrCompare + '(Buffer)');
```

Comment:
Do we need a second argument just for throwing errors which should never happen?